### PR TITLE
Create search-bar component

### PR DESCRIPTION
**Pull request by alyxrossetti**
@rit I started looking at type ahead components but decided we should probably discuss before choosing one. @joshuago78 said type ahead was not a requirement for this card.

### CODE REVIEW

**Comment by alyxrossetti** [comment]
@rit Conflicts have been resolved and tests have been added - this PR is ready for review

File: gpi/frontend/web/src/ui/components/layout/base.vue [hunk 1]
```diff
@@ -20,6 +18,7 @@
 import HeaderExplore from '@ui/header-explore.vue'
 import HeaderLogo from '@ui/header-logo.vue'
 import HeaderNav from '@ui/header-nav'
+import SearchBar from '@ui/search-bar.vue'
```
Comment (rit) [inline gpi/frontend/web/src/ui/components/layout/base.vue]: With core.js, we no longer need this line.

File: gpi/frontend/web/src/ui/components/layout/base.vue [hunk 2]
```diff
@@ -28,6 +27,7 @@ export default {
     HeaderExplore,
     HeaderLogo,
     HeaderNav,
+    SearchBar,
```
Comment (rit) [inline gpi/frontend/web/src/ui/components/layout/base.vue]: To ease the testing with Cypress, we will register components globally with `core.js` therefore we should stop using the `components` attribute for our components.

File: gpi/frontend/web/src/ui/components/search-bar.vue [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          </div>
+        </div>
+
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import FontAwesomeIcon from '@fortawesome/vue-fontawesome'
```
Comment (rit) [inline gpi/frontend/web/src/ui/components/search-bar.vue]: All icons related stuff is handled in the `icons.js` file

File: gpi/frontend/web/src/ui/components/search-bar.vue [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+export default {
+  name: 'SearchBar',
+  components: {
```
Comment (rit) [inline gpi/frontend/web/src/ui/components/search-bar.vue]: This line is no longer needed.

File: gpi/frontend/web/src/ui/components/layout/base.vue (hunk 2 above)
Comment (alyxrossetti) [reply gpi/frontend/web/src/ui/components/layout/base.vue]: Got it. Done.

**Comment by alyxrossetti** [comment]
@rit I did get it - just wasn't sure if you wanted them all removed in this branch. Looks good to me! Merging.

**Pull request merged**